### PR TITLE
Use `FilesToRunProvider.executable` to find the Swift worker executable.

### LINKED ISSUE #164: Update `spm_xcode_toolchain.bzl` to use `DefaultInfo.files_to_run`.
[Related Slack post](https://bazelbuild.slack.com/archives/CD3QY5C2X/p1662053486433759?thread_ts=1662053013.300319&cid=CD3QY5C2X)
> I believe https://github.com/cgrindel/rules_spm/blob/0cbd68127a333a7b39a25f5175ca189ee017882f/spm/private/spm_xcode_toolchain.bzl#L28 needs to use DefaultInfo.files_to_run similar to what I had to do in that commit.

### PR DESCRIPTION
**Pull request by cgrindel**
Closes #164.

### CODE REVIEW

File: spm/private/spm_linux_toolchain.bzl [hunk 1]
```diff
@@ -21,7 +21,8 @@ Could not parse the version number for Swift package manager. {version}\
     return current_version >= desired_version_value
 
 def _create_build_tool_config(ctx, target_triple, spm_configuration):
-    swift_worker = ctx.executable._swift_worker
+    swift_worker = ctx.attr._swift_worker[DefaultInfo].files_to_run.executable
```
Comment (brentleyjones) [inline spm/private/spm_linux_toolchain.bzl:24]: The `.executable` at the end makes this a no-op change. You want to pass the provider to `additional_tools`. You might need to use `.executable` in `args` though.

**Review by brentleyjones — COMMENTED** [review]
I would test with this commit to confirm your change: https://github.com/bazelbuild/rules_swift/commit/8aca87eba7dc9164c9835b887ff58d1c8ddbf20e

File: spm/private/spm_linux_toolchain.bzl (hunk 1 above)
Comment (cgrindel) [reply spm/private/spm_linux_toolchain.bzl]: You are correct. I have updated the code to use the `.executable` for the arg, but I pass along the `FilesToRunProvider` to `additional_tools` which, in turn, is passed to `execution_requirements` for `ctx.actions.run()`.

I have run integration tests specifically referencing [the commit that you suggested](https://github.com/bazelbuild/rules_swift/commit/8aca87eba7dc9164c9835b887ff58d1c8ddbf20e). I believe that the runfiles are being propagated properly.

File: spm/private/spm_xcode_toolchain.bzl [hunk 2]
```diff
@@ -138,10 +138,10 @@ The location for the utilities that are required by SPM.\
 """,
         ),
         "_swift_worker": attr.label(
-            cfg = "host",
+            cfg = "exec",
             allow_files = True,
             default = Label(
-                "@build_bazel_rules_swift//tools/worker",
+                "@build_bazel_rules_swift//tools/worker:worker_wrapper",
```
Comment (brentleyjones) [inline spm/private/spm_xcode_toolchain.bzl:144]: Why was this change needed?

**Review by brentleyjones — APPROVED** [review]

File: spm/private/spm_xcode_toolchain.bzl (hunk 2 above)
Comment (cgrindel) [reply spm/private/spm_xcode_toolchain.bzl:144]: Not sure that it was. I noticed that `rules_swift` was using it. The target is an `alias` that resolves to `//tools/worker:worker` or `//tools/worker:universal_worker`.
